src/ol/render/Feature.js

```diff
@@ -326,4 +337,76 @@ RenderFeature.prototype.getEndss = RenderFeature.prototype.getEnds;
 RenderFeature.prototype.getFlatCoordinates =
   RenderFeature.prototype.getOrientedFlatCoordinates;
 
+/**
+ * Create a geometry from an `ol/render/Feature`
+ * @param {RenderFeature} renderFeature
+ * Render Feature
+ * @return {Point|MultiPoint|LineString|MultiLineString|Polygon|MultiPolygon}
+ * New geometry instance.
+ * @api
+ */
+export function toGeometry(renderFeature) {
+  const geometryType = renderFeature.getType();
+  switch (geometryType) {
+    case GeometryType.POINT:
+      return new Point(renderFeature.getFlatCoordinates());
+    case GeometryType.MULTI_POINT:
+      return new MultiPoint(
+        renderFeature.getFlatCoordinates(),
+        GeometryLayout.XY
+      );
+    case GeometryType.LINE_STRING:
+      return new LineString(
+        renderFeature.getFlatCoordinates(),
+        GeometryLayout.XY
+      );
+    case GeometryType.MULTI_LINE_STRING:
+      return new MultiLineString(
+        renderFeature.getFlatCoordinates(),
+        GeometryLayout.XY,
+        /** @type {Array<number>} */ (renderFeature.getEnds())
+      );
+    case GeometryType.POLYGON:
+      const flatCoordinates = renderFeature.getFlatCoordinates();
+      const ends = /** @type {Array<number>} */ (renderFeature.getEnds());
+      const endss = inflateEnds(flatCoordinates, ends);
+      return endss.length > 1
+        ? new MultiPolygon(flatCoordinates, GeometryLayout.XY, endss)
+        : new Polygon(flatCoordinates, GeometryLayout.XY, ends);
+    case GeometryType.MULTI_POLYGON:
```

Comment:
You can remove this case. There is no `GeometryType.MULTI_POLYGON` for `ol/render/Feature`. See `ol/format/MVT.getGeometryType()`. For rendering, all that matters is whether the ring is counterclockwise (outer ring) or clockwise (inner ring). Because of the order of the rings, we know that all clockwise rings following a counterclockwise ring are inner rings of that counterclockwise ring. When the next counterclockwise ring is encountered, it is the outer ring of another polygon.

Reply:
I wasn't sure whether a MultiPolygon was used anywhere, because it should work, though less efficiently.
It is removed now.